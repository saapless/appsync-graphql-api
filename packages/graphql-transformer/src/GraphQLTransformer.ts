export class GraphQLTransformer {
  constructor() {}

  public transform() {
    return "transformed";
  }
}
