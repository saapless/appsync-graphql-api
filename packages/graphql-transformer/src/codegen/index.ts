export * from "./ast";
export * from "./printer";
export * from "./document";
