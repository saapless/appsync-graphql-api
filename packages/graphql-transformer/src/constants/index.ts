export const ConnectionDirective = Object.freeze({
  HAS_ONE: "hasOne",
  HAS_MANY: "hasMany",
});

export const ModelDirective = Object.freeze({
  MODEL: "model",
});

export const UtilityDirective = Object.freeze({
  SERVER_ONLY: "serverOnly",
  CLIENT_ONLY: "clientOnly",
  READ_ONLY: "readOnly",
  WRITE_ONLY: "writeOnly",
  FILTER_ONLY: "filterOnly",
  INTERNAL: "internal",
});

export const ScalarType = Object.freeze({
  STRING: "String",
  INT: "Int",
  FLOAT: "Float",
  BOOLEAN: "Boolean",
  ID: "ID",
  LONG: "Long",
});

export const RelationType = Object.freeze({
  ONE_TO_ONE: "oneToOne",
  ONE_TO_MANY: "oneToMany",
  MANY_TO_MANY: "manyToMany",
});
