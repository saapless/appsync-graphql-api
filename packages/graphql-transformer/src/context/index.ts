export * from "./TransformerContext";
export type {
  DescriptorOperation,
  Key,
  KeyOperator,
  KeyValue,
  OperationActionType,
  ResolverDescriptor,
  ReturnType,
} from "./ResolverLoader";
