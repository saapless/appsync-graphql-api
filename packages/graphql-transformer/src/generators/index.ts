export { ResolverGeneratorBase } from "./ResolverGeneratorBase";
export { DynamoDbGenerator } from "./DynamoDBGenerator";
export { NoneGenerator } from "./NoneGenerator";
export { SchemaTypesGenerator } from "./SchemaTypesGenerator";
export { ResolverTypesGenerator } from "./ResolverTypesGenerator";
