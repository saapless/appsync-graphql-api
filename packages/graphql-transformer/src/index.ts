export * from "./transformer";
export { definitionFromSource } from "./utils";
