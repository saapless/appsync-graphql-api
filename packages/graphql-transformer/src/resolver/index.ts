export { ResolverBase } from "./ResolverBase";
export { FieldResolver } from "./FieldResolver";
export { FunctionResolver } from "./FunctionResolver";
