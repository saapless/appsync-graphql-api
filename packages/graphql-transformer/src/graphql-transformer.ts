import { GraphQLTransformer } from "./GraphQLTransformer";

export function createTransformer() {
  return new GraphQLTransformer();
}

export function executeTranformer() {}
