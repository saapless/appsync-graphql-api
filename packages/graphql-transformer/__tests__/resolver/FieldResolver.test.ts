import { FieldResolver, _call, _chain, _id, _obj, _prop, _return } from "../../src/resolver";

describe("resolver/FieldResolver", () => {
  const resolver = FieldResolver.create("Query", "node");
  it("should create resolver code", () => {
    resolver
      .addImport("@aws-appsync/utils", "util")
      .addImport("@aws-appsync/utils/dynamodb", "get")
      .setRequest(
        _return(_call(_id("get"), [_obj(_prop("key", _obj(_prop("id", _chain("ctx.args.id")))))]))
      );

    expect(resolver.serialize()).toMatchSnapshot();
  });

  describe("given a source", () => {
    it("parse resolver from source code", () => {
      const fromSource = FieldResolver.fromSource("Query", "user", "function request() {}");
      expect(fromSource.isReadonly).toEqual(true);
    });
  });
});
