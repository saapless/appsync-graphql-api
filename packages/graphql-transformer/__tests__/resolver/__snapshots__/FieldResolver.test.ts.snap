// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolver/FieldResolver should create resolver code 1`] = `
{
  "code": "",
  "dataSource": undefined,
  "fieldName": "node",
  "pipelineFunctions": undefined,
  "typeName": "Query",
}
`;
